src/obs_inv_utils/aws_s3_interface.py

```diff
@@ -0,0 +1,243 @@
+import re
+import attr
+from collections import namedtuple, OrderedDict
+import boto3
+from datetime import datetime
+from botocore.config import Config
+from botocore import UNSIGNED
+
+
+bdp_config = Config(
+    region_name = 'us-east-1',
+    signature_version = UNSIGNED,
+    retries = {
+        'max_attempts': 10,
+        'mode': 'standard'
+    }
+)
+
+nl = '\n'
+
+AwsS3ObjectsListContents = namedtuple(
+    'AwsS3ObjectsListContents',
+    [
+        'prefix',
+        'files_count',
+        'listed_objects',
+        'obs_cycle_time',
+        'submitted_at',
+        'latency'
+    ],
+)
+
+AwsS3CommandRawResponse = namedtuple(
+    'AwsS3CommandRawResponse',
+    [
+        'command',
+        'return_code',
+        'output',
+        'success',
+        'args_0',
+        'submitted_at',
+        'latency'
+    ],
+)
+
+
+AwsS3FileMeta = namedtuple(
+    'AwsS3FileMeta',
+    [
+        'name',
+        'permissions',
+        'last_modified',
+        'size'
+    ],
+)
+
+
+AwsS3Command = namedtuple(
+    'AwsS3Command',
+    [
+        'command',
+        'arg_validator',
+        'output_parser'
+    ],
+)
+
+AWS_BDP_BUCKET = 'noaa-reanalyses-pds'
+CMD_GET_S3_OBJ_LIST = 'list_objects'
+
+
+def get_bdp_s3_client():
+    try:
+        client = boto3.client('s3', config=bdp_config)
+    except Exception as e:
+        msg = f'Problem getting boto3 s3 client - error: {e}'
+        raise ValueError(msg)
+
+    return client
+
+
+def get_s3_objects_list(client, bucket=None, prefix=None):
+    try:
+        response = client.list_objects_v2(Bucket=bucket, Prefix=prefix)
+    except Exception as e:
+        msg = f'Problem getting list for prefix: {prefix}, error: {e}'
+        print(msg)
+
+    return response
+        
+
+def get_objects_list_args_valid(args):
+    if not isinstance(args, list):
+        msg = f'Args must be in the form of a list, args: {args}'
+        raise TypeError(msg)
+    cmd = aws_s3_cmds[CMD_GET_S3_OBJ_LIST].command
+    print(f'{nl}{nl}In inspect tarball args valid: cmd: {cmd}{nl}{nl}')
+    if (len(args) > 1 or len(args) == 0):
+        msg = f'Command "{cmd}" accepts exactly 1 argument, received ' \
+              f'{len(args)}.'
+        raise ValueError(msg)
+
+    arg = args[0]
+
+    try:
+        m = re.search(r'[^A-Za-z0-9\._\-\/]', arg)
+        if m is not None and m.group(0) is not None:
+            print('Only a-z A-Z 0-9 and - . / _ characters allowed in filepath')
+            raise ValueError(f'Invalid characters found in file path: {arg}')
+    except Exception as e:
+        raise ValueError(f'Invalid file path: {e}')
+
+    return True
+
+
+def s3_object_list_v2_parser(obj_list_contents, obs_cycle_time):
```

Comment:
why there is a "v2" string in the name. does it mean version 2?

Reply:
That is a carryover from the boto3 function definition.  I can leave it out if it seems confusing.

Reply:
Yes, it is directly inherited from boto3.